backend/indexer/indexer.py

```diff
@@ -364,8 +371,18 @@ async def ingest_data(request: IngestDataToCollectionDto):
                         raise_error_on_failure=created_data_ingestion_run.raise_error_on_failure,
                         batch_size=request.batch_size,
                     )
-                )
-                created_data_ingestion_run.status = DataIngestionRunStatus.COMPLETED
+                # Import here to avoid circular dependency
+                from backend.server.app import process_pool
+                # future of this submission is ignored, ingestion failure due to process termination will not be tracked
```

Comment:
Let's not cross import
it is fine to pass `pool` as input arg which is by default `None`

Reply:
We want to keep all the code as is and rather use something on lines of 

```
loop = asyncio.get_running_loop()
asyncio.create_task(loop.run_in_executor(sync_data_source_to_collection(...), pool, ...))
```

Reply:
@chiragjn [run_in_executor](https://docs.python.org/3/library/asyncio-eventloop.html#asyncio.loop.run_in_executor) is  used to run synchronous code, so sync_data_source_to_collection can't be async
further, asyncio.create_task(coro) will again block the event loop (waiting for coro to finish at some point)

Reply:
> [run_in_executor](https://docs.python.org/3/library/asyncio-eventloop.html#asyncio.loop.run_in_executor) is used to run synchronous code

You are correct, I am not framing it correctly, give me some time, I'll get back on this

Reply:
@chiragjn I spend a few hours thinking about it too, I know you want to run the whole ingestion in the same async event loop, but there are blocking synchronous parts in the sync_data_source_to_collection function, mainly the getting of chunks from unstructured_api, which will keep blocking other tasks in the event loop.
   
   this method get_chunks in 
   backend.modules.parsers.unstructured_io.UnstructuredIoParser
   does a very long POST call which blocks everything
   
if we make somehow make this request non-blocking, we would be in a better shape, but there still wont be any guarantee that it doesn't block the main event loop.

Reply:
> Let's not cross import it is fine to pass `pool` as input arg which is by default `None`

@chiragjn add a process_pool module to make this cleaner.

Reply:
> I know you want to run the whole ingestion in the same async event loop,

Actually I don't want to in run in same loop. I want to run stuff in a separate process, my only hesistation is wrapping all async coroutines to be sync again, just trying to see if we can somehow avoid that. Worst case we have to take this route

> this method get_chunks in
> backend.modules.parsers.unstructured_io.UnstructuredIoParser
> does a very long POST call which blocks everything
>
> if we make somehow make this request non-blocking, we would be in a better shape, but there still wont be any guarantee that it doesn't block the main event loop.

Yes this and plus the whole vector store implementation also needs to be made async.
At least the unstructured API call should be easy to change, instead of requests we just have to use asyncio httpx or  aiohttp

> but there still wont be any guarantee that it doesn't block the main event loop.

That is true, some bad parser can still block, hence eventually we will move away to a queue and workers system.